en_US/extensions/plugins.md

```diff
@@ -1,43 +1,73 @@
 # Plugins
 
-EMQX allows users to customize the business logic or implement other protocols using plugins written in Erlang. This page introduces how to develop a customized plugin.
+EMQX enables users to customize business logic or implement additional protocols by creating plugins written in Erlang. This guide covers the process of developing, installing, and managing plugins for EMQX.
 
-The basic process of plugin development and operation is as follows:
+## What is a Plugin?
 
-- Download and install our [rebar3 emqx-plugin template](https://github.com/emqx/emqx-plugin-template).
-- Generate the corresponding plugin tarball using the plugin template provided by EMQX.
-- Install the plugin package via Dashboard or Command Line Interface (CLI).
-- Start/stop/uninstall your plugin via Dashboard or CLI.
+A plugin is an Erlang application that operates within the EMQX nodes.
 
-:::tip Prerequisite
+To be loaded into the nodes, a plugin must be compiled into a release (a single `.tar.gz` file). This release can then be imported into EMQX using the Dashboard, REST API, or CLI interfaces.
 
-Knowledge of EMQX [hooks](./hooks.md)
-:::
+Once loaded, a plugin can be:
+
+- Configured
+- Started or stopped
+- Unloaded
+
+Upon startup, a plugin typically registers some of its functions as EMQX *callbacks* to modify or extend the behavior of EMQX.
 
 ## Develop EMQX Plugins
 
-This section provides a step-by-step guide on developing custom EMQX plugins, using the creation of an access control plugin as an example.
+This section provides a step-by-step guide to developing custom EMQX plugins, using the creation of an access control plugin as an example.
 
-### Download and Install the rebar3 Plugin Template
+### Prerequisites
 
-EMQX offers an [emqx-plugin-template](https://github.com/emqx/emqx-plugin-template) to facilitate the creation of custom emqx-plugin projects.
+Before you begin, make sure you have the following:
 
-1. Use the following commands to download the `emqx-plugin-template`:
+- Knowledge of EMQX [hooks](./hooks.md).
+- A working build environment (e.g., `build_essential`), including `make`.
+- Erlang/OTP of the same major version as the EMQX release you wish to target. For more information, see the `org.opencontainers.image.otp.version` attribute in the Docker or refer to the `.tool-versions` file for the used version (e.g., https://github.com/emqx/emqx/blob/e5.9.0-beta.4/.tool-versions). It's recommended to use [ASDF](https://asdf-vm.com/) to manage Erlang/OTP versions.
+- [rebar3](https://www.rebar3.org/).
```

Comment:
maybe advise to pull emqx-builder image as an alternative.
run this command to list all image tags: https://github.com/emqx/emqx-builder/blob/main/show-latest-images.sh